src/mediapipe_internal/pythonnoderesource.hpp

```diff
@@ -20,24 +20,26 @@
 #if (PYTHON_DISABLE == 0)
 #include <pybind11/embed.h>  // everything needed for embedding
 
-#include "src/mediapipe_calculators/python_backend_calculator.pb.h"
+#include "src/mediapipe_calculators/python_executor_calculator_options.pb.h"
 
 namespace py = pybind11;
 #endif
 
 namespace ovms {
 class Status;
+class PythonBackend;
 
 class PythonNodeResource {
 public:
     PythonNodeResource(const PythonNodeResource&) = delete;
     PythonNodeResource& operator=(PythonNodeResource&) = delete;
 #if (PYTHON_DISABLE == 0)
-    PythonNodeResource();
-    PythonNodeResource(const google::protobuf::Any& nodeOptions, Status& status);
     std::unique_ptr<py::object> nodeResourceObject;
+    PythonBackend * pythonBackend;
```

Comment:
Why are those fields public? Either switch PythonNodeResource to be struct if you want to expose all the fields or add getters and setters. Class purpose should be to encapsulate some behavior, hide implementation details and provide an api

Reply:
I think switching to private will be better.